### ISSUE TITLE
Open first incomplete section when Getting Started shows

### ISSUE THREAD

**Issue by digitarald** [microsoft/vscode#116570]
User research showed that many users just scanned the page to determine if it is useful for them. Only after clicking into the sections, some users got their aha moment.

Reducing choices for new users, the first incomplete (not all sections unchecked) section should open. This helps both for Setup and the content presented in Codespaces.

Alternatively, if the first section is partially done, we could show the index and let the user pick if they like to continue the partial section or a new one. @misolori any thoughts?

This assumes that items are easier to check off (e.g. not making users change their theme if they don't want to), so more issues to be filed on that.

**Comment by misolori** [comment]
I agree that if users aren't clicking into the content for various reasons but do find the content useful we should open the first item. I do think we'll need to do some tweaking to make this smoother, here's what it would look like:

![image](https://user-images.githubusercontent.com/35271042/107803963-2e65d380-6d18-11eb-9a60-001fd44edbff.png)

Things that stand out:
- It's weird that there is a back button on this landing page
- There is no centered title that tells me what this is
- The title "Quick Setup" and captions don't really give me a welcoming entrance, I think we'd need a friendlier "Welcome to VS Code" vibe
- Maybe there should be a link somewhere to see "more content" if they want (this is what the back button does)

**Comment by digitarald** [comment]
> There is no centered title that tells me what this is
> The title "Quick Setup" and captions don't really give me a welcoming entrance, I think we'd need a friendlier "Welcome to VS Code" vibe

Right, if we show this first, the initial content should be adapted to give a warm welcome.

> Maybe there should be a link somewhere to see "more content" if they want (this is what the back button does)

Totally, this is a dead right now! Filed #116574 for that.

**Comment by JacksonKearl** [comment]
One alternate idea would be to elevate some of the most popular items to the front page, for instance: 
![image](https://user-images.githubusercontent.com/8586769/107828220-3dab4800-6d3d-11eb-993a-1d62d7af5640.png)
This brings the value propositions a bit more front and center for the most important items, rather than needing to click into lists.

**Comment by digitarald** [comment]
> One alternate idea would be to elevate some of the most popular items to the front page, for instance:

This would make sense if we always want to make users aware of the same features. I'd be concerned that these items work different from the todo-like entries in the lists and can never get checked off. Showing top items + all sections also doesn't fix the problem of giving the user too many choices.

**Comment by digitarald** [comment]
Summarizing an argument from @kieferrm: When we add more content areas that users can pick between (beginners/advanced, languages, extensions, etc), opening the first incomplete section might not be the right content for for the user.

One solution that keeps this in mind would be to keep the first section as widely applicable as possible (which would be Setup in case of VS Code) and still open it by default. After this initial section is partially completed, all future sessions open on the index. This way we can reduce choice fatigue for new users and still ensure that the rest of the content can be discovered.

The flow of opening Getting Started for languages & extensions needs generally more discussions – for example when it will be presented to users after they install the extension.

**Comment by JacksonKearl** [comment]
Opening the first section when there have been no completed items makes sense to me. 👍

**Comment by JacksonKearl** [comment]
Will now open to the first section if no items have been completed. Also, the `Back` button has been renamed to `More` to better fit this situation. 

The opening screen for new users now looks like this:
![image](https://user-images.githubusercontent.com/8586769/108410315-1d1b3c00-71dc-11eb-997a-ed45776c5300.png)

This could be more welcoming. Will create a new issue to track.

**Comment by kieferrm** [comment]
@digitarald regarding https://github.com/microsoft/vscode/issues/116570#issuecomment-780315690: I don't think that having a generally applicable section really helps. On the contrary, I'd expect that to lead to more dismissals. So, we should be sure that we user test this appropriately.

**Comment by digitarald** [comment]
We got this as an idea from usertesting on new users, but I can see the point in not hiding the full range of content from experienced users. @JacksonKearl to address @kieferrm's point, could we put the logic from f4ce983bcd70bf8ed0da1cad4abc9cb9fd138b05 behind an experiment feature flag to do more testing in the field? Shall I file a new issue?

**Comment by kieferrm** [comment]
That would make sense to me.
